Sources/StreamChatUI/ChatMessageList/Attachments/ChatFileAttachmentListView+ItemView.swift

```diff
@@ -80,72 +75,64 @@ extension _ChatMessageFileAttachmentListView {
-                ),
-                fileNameAndSizeStack.centerYAnchor.pin(equalTo: centerYAnchor),
-                fileNameAndSizeStack.topAnchor.pin(greaterThanOrEqualTo: layoutMarginsGuide.topAnchor),
-                fileNameAndSizeStack.bottomAnchor.pin(lessThanOrEqualTo: layoutMarginsGuide.bottomAnchor)
-            ])
+            addSubview(mainContainerStackView)
+            mainContainerStackView.pin(to: layoutMarginsGuide)
+            
+            spinnerAndSizeStack.addArrangedSubviews([loadingIndicator, fileSizeLabel])
+            fileNameAndSizeStack.addArrangedSubviews([fileNameLabel, spinnerAndSizeStack])
+            mainContainerStackView.addArrangedSubviews([fileIconImageView, fileNameAndSizeStack, actionIconImageView])
+            
+            spinnerAndSizeStack.axis = .horizontal
+            spinnerAndSizeStack.spacing = 8
+            spinnerAndSizeStack.alignment = .leading
+            
+            fileNameAndSizeStack.axis = .vertical
+            fileNameAndSizeStack.alignment = .leading
+            fileNameAndSizeStack.spacing = 3
+            
+            mainContainerStackView.axis = .horizontal
+            mainContainerStackView.spacing = 8
+            mainContainerStackView.alignment = .center
         }
 
         override open func updateContent() {
             super.updateContent()
 
             fileIconImageView.image = fileIcon
-            fileNameLabel.text = content?.type.rawValue
+            // If we cannot fetch filename, let's use only content type.
+            fileNameLabel.text = content?.payload.title ?? content?.type.rawValue
 
-            if case .uploadingFailed = content?.uploadingState?.state {
+            switch content?.uploadingState?.state {
+            case .uploaded:
+                fileSizeLabel.text = content?.payload.file.sizeString
+            case .uploadingFailed:
                 fileSizeLabel.text = L10n.Message.Sending.attachmentUploadingFailed
-            } else {
+            default:
                 fileSizeLabel.text = content?.uploadingState?.fileUploadingProgress
             }
-
-            if let state = content?.uploadingState?.state {
-                actionIconImageView.image = appearance.fileAttachmentActionIcon(for: state)
-            } else {
+        
+            switch content?.uploadingState?.state {
+            case .none:
                 actionIconImageView.image = nil
+            default:
+                actionIconImageView.image = appearance.fileAttachmentActionIcon(for: content!.uploadingState!.state)
             }
 
             switch content?.uploadingState?.state {
             case .pendingUpload, .uploading:
-                loadingIndicator.isVisible = true
+                spinnerAndSizeStack.insertSubview(loadingIndicator, at: 0)
             default:
-                loadingIndicator.isVisible = false
+                spinnerAndSizeStack.removeArrangedSubview(loadingIndicator)
             }
```

Comment:
`updateContent` can be called multiple times with the same content. Using `isVisible` flag works better in this case

Reply:
Thanks, you are right, reverted :)